### PR TITLE
(569b) Restore E2E tests

### PR DESCRIPTION
**Pull request by yndajas**
## Context

In f74fcf7 we disabled the E2E tests due to mismatches with the data on dev, which were undergoing frequent changes. The data are now more stable, so we're ready to re-enable the E2E tests once fixed (in a [related PR][1])

## Changes in this PR

- Re-enable E2E test step in CI

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [x] There are changes required to the Accredited Programmes API for this change to work...
  - [x] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [x] This makes new expectations of the API...
  - [x] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

[1]: https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e/pull/3

### CODE REVIEW

**Comment by yndajas** [comment]
[Re-enable E2E CI step](https://trello.com/c/akwYQCVu/569-re-enable-e2e-ci-step)

**Review by Harriethw — APPROVED** [review]
looks good! 
Have you thought about sending an alert to Slack if the tests fail? [That's what we're doing in CAS](https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/blob/65d4475db029a0f1d4e7244d57b1096c57565907/.circleci/config.yml#L83C3-L83C3)
```
      - slack/notify:
          event: fail
          channel: << pipeline.parameters.alerts-slack-channel >>
          template: basic_fail_1
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -150,6 +150,27 @@ jobs:
...
+  end_to_end_test:
+    docker:
+      - image: mcr.microsoft.com/playwright:v1.34.2-focal
+    circleci_ip_ranges: true # opt-in to jobs running on a restricted set of IPs
+    steps:
+      - run:
+          name: Clone E2E repo
+          command: |
+            git clone https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e.git .
+      - run:
```
Comment (richpjames) [inline .circleci/config.yml:162]: v minor but is installing the latest version of `npm` necessary?

**Review by richpjames — APPROVED** [review]
Looks good, one minor comment - feel free to ignore

**Comment by yndajas** [comment]
> looks good! Have you thought about sending an alert to Slack if the tests fail? [That's what we're doing in CAS](https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/blob/65d4475db029a0f1d4e7244d57b1096c57565907/.circleci/config.yml#L83C3-L83C3)
> 
> ```
>       - slack/notify:
>           event: fail
>           channel: << pipeline.parameters.alerts-slack-channel >>
>           template: basic_fail_1
> ```

That's a good idea. We do have Slack alerts for deployments to prod, but not other steps. I think this is probably one to discuss with George separately when he's back though

File: .circleci/config.yml (hunk 1 above)
Comment (yndajas) [reply .circleci/config.yml:162]: So it looks like it came from https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/50/commits/ad97f5abffcfe87ce6b88ef1451a39aff7b3493b, which I'd imagine is copying from Approved Premises, e.g. https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/173/commits/9633dcc4a4a8c6b7a35c0d178988151cf5d3ce99 and https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/670/commits/35fadeaa33611f7b8f4bdf93bc60e6d51f0174d4

I think we might need to update the artifacts path though: https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/904/commits/7aee650ae4d24f97dc4e88587944cddde7180994

**Pull request merged**